### PR TITLE
[matrix] Enable Transpose for rectangular matrices

### PR DESCRIPTION
**Pull request by mdessole**
# This Pull request:

Enables Transpose to work with rectangular matrices. SetMatrixArray has been modified to reshape the matrix, if needed. A Google Test has been added to test this functionality.

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

This PR fixes [ROOT-9261](https://its.cern.ch/jira/browse/ROOT-9261)

### CODE REVIEW

**Review by lmoneta — APPROVED** [review]
LGTM!
Thank you for adding this missing feature. 
One comment I have is maybe organize better the commits, having one for the new functionality and the documentation and one on the formatting and one on the tests

File: math/matrix/inc/TMatrixTSparse.h [hunk 1]
```diff
@@ -31,176 +31,257 @@
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
-template<class Element> class TMatrixT;
+template <class Element>
```
[the remark this answers is not in the document]
Comment (mdessole) [reply math/matrix/inc/TMatrixTSparse.h:34]: I rearranged the commits in order to make change tracking easier.

**Comment by hahnjo** [comment]
@mdessole I cancelled the CI jobs on `mac12` and `mac13` - they were stuck since 3 hours in the CMake configuration step (same result as previous runs that timed out after 6 hours). Maybe try rebasing the PR with the latest CI fixes in `master`?

**Comment by hahnjo** [comment]
I cancelled the stuck `mac12` and `mac13` once more...

**Pull request merged**